### PR TITLE
Add file type checking for CUE spec compliance

### PR DESCRIPTION
**Pull request by benjaminglass1**
Solves issue #226 . Adds a fairly rudimentary file extension checking system that should write the correct comments into the CUE files.

### CODE REVIEW

**Review by zas — CHANGES_REQUESTED** [review]
Looks ok to me but you should increase plugin version number too.

**Comment by benjaminglass1** [comment]
@zas I amended the commit to add that. :+1:

**Review by zas — APPROVED** [review]

File: plugins/cuesheet/cuesheet.py [hunk 1]
```diff
@@ -184,7 +184,12 @@ def callback(self, objs):
                     audio_filename = file.filename
                     if os.path.dirname(filename) == os.path.dirname(audio_filename):
                         audio_filename = os.path.basename(audio_filename)
-                    cuesheet.tracks[i].set("FILE", audio_filename, "MP3")
+                    if audio_filename.split(".")[-1].lower() in ["mp3", "mp2", "m2a"]:
```
Comment (zas) [inline plugins/cuesheet/cuesheet.py]: You can perhaps reduce the code redundancy a bit, with something like:

```python
extension = audio_filename.split(".")[-1].lower()
if extension in ["mp3", "mp2", "m2a"]:
    filetype = "MP3"
elif extension in ["aiff", "aif", "aifc"]:
    filetype = "AIFF"
else:
   filetype = "WAV"
cuesheet.tracks[i].set("FILE", audio_filename, filetype)
```

Comment (benjaminglass1) [reply plugins/cuesheet/cuesheet.py]: Got it. :+1:

**Review by zas — APPROVED** [review]

File: plugins/cuesheet/cuesheet.py [hunk 2]
```diff
@@ -182,9 +182,16 @@ def callback(self, objs):
...
+                    extension = audio_filename.split(".")[-1].lower()
                     if os.path.dirname(filename) == os.path.dirname(audio_filename):
                         audio_filename = os.path.basename(audio_filename)
-                    cuesheet.tracks[i].set("FILE", audio_filename, "MP3")
+                    if extension in ["mp3", "mp2", "m2a"]:
+                        file_type = "MP3"
+                    elif extension in ["aiff", "aif", "aifc"]:
+                        file_type = "AIFF"
+                    else:
+                        file_type = "WAV"
```
Comment (phw) [inline plugins/cuesheet/cuesheet.py:193]: @benjaminglass1 Sorry for jumping in here so late, but this part puzzles me a bit. Why does it set everything else to WAV? I kind of get how this makes sense for FLAC (equaling WAV with lossless), but does it also make sense for thing like WMA and MP4?

Comment (benjaminglass1) [reply plugins/cuesheet/cuesheet.py:193]: According to the [archived cue sheet spec](https://web.archive.org/web/20160201021136/http://digitalx.org/cue-sheet/syntax/) and the HydrogenAudio wiki page for [cue sheets](http://wiki.hydrogenaud.io/index.php?title=Cue_sheet#Most_often_used), the acceptable 5 file types are WAVE (which I just realized I misspelled [facepalm]), MP3, AIFF, BINARY and MOTOROLA. BINARY and MOTOROLA were both specifically for some form of binary format that afaik doesn't exist anymore, which means all audio files get stuck in the other 3 categories. Based on the HydrogenAudio wiki, it appears to be the convention that files not fitting neatly into MP3 or AIFF should be categorized as WAVE.

Comment (phw) [reply plugins/cuesheet/cuesheet.py:193]: Ok, got it. Thanks for the clarification :) Will you submit a PR for the `WAVE` fix? I can also just push this minor fix directly if you prefer.

**Comment by benjaminglass1** [comment]
I'd rather just have you push the fix directly, if you don't mind.




On Fri, Aug 9, 2019 at 7:34 AM -0500, "Philipp Wolfer" <notifications@github.com> wrote:












@phw commented on this pull request.





In plugins/cuesheet/cuesheet.py:
>                      if os.path.dirname(filename) == os.path.dirname(audio_filename):
                         audio_filename = os.path.basename(audio_filename)
-                    cuesheet.tracks[i].set("FILE", audio_filename, "MP3")
+                    if extension in ["mp3", "mp2", "m2a"]:
+                        file_type = "MP3"
+                    elif extension in ["aiff", "aif", "aifc"]:
+                        file_type = "AIFF"
+                    else:
+                        file_type = "WAV"



Ok, got it. Thanks for the clarification :) Will you submit a PR for the WAVE fix? I can also just push this minor fix directly if you prefer.

File: plugins/cuesheet/cuesheet.py (hunk 2 above)
Comment (benjaminglass1) [reply plugins/cuesheet/cuesheet.py:193]: Push it directly, if you don't mind. Seems too small for a full pull request.

Comment (phw) [reply plugins/cuesheet/cuesheet.py:193]: Done in https://github.com/metabrainz/picard-plugins/commit/70b797d139d8d1a5089681e4530c4d04dda031ee

**Pull request merged**